src/jit/morph.cpp

```diff
@@ -12461,7 +12461,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                         // We need to keep op1 for the side-effects. Hang it off
                         // a GT_COMMA node
 
-                        tree->ChangeOper(GT_COMMA);
+                        tree->ChangeOper(GT_COMMA, GenTree::PRESERVE_VN);
```

Comment:
I believe that this is incorrect:

We have a subtree that is  this
GT_MUL(op1, CNS(0))

And op1 has some side effect that we want to keep.

Currently the VN for the GT_MUL represents (op1 * 0) with the execeptional effects from op1.

The new GT_COMMA node should get a different value number that what the GT_MUL had.